docs/development.md

```diff
@@ -1,6 +1,260 @@
+ 
+  This is a little bit complicated and error prone because it is splitted into three steps:
+  
+  - Parse YAML file
+  - Convert YAML config structs to runtime test structs
+  - Assign global configuration
+  
+  For this take a look at the `pkg/suite/yaml_suite.go:ParseYAML` function which is responsible for parsing the suite.
+  
+  1. Parse yaml - `err := yaml.UnmarshalStrict(content, &yamlConfig)`
+      This line parses the yaml file and will return a `suite.YAMLSuiteConf`, later it will be converted to our structs
+      from the `runtime` package.
+      Now we can implement our custom unmarshal logic in the `YAMLSuiteConf::UnmarshalYAML` method.
+      Follow the code where `Config:   y.mergeConfigs(v.Config, params.Config)` is called. 
+      This merges the local with global configs and overwrites the global if it is necessary.
+      
+      Jump into the `YAMLSuiteConf::mergeConfigs` and add the `message` property like this:
+      
+      ```go
+      if local.Message != "" {
+          conf.Message = local.Message
+      }
+
+      ```
+
+  1. Convert test cases - `tests := convertYAMLSuiteConfToTestCases(yamlConfig)`
+  
+     Jump into the `convertYAMLSuiteConfToTestCases` function and assign the content of our new field to the `runtime.CommandUnderTest` conversion.
+     
+     ```
+      Command: runtime.CommandUnderTest{
+          Cmd:        t.Command,
+          InheritEnv: t.Config.InheritEnv,
+          [...]
+          Message:    t.Config.Message,
+      },
+      ```
+      
+**5. Add the global config assignment and add the `Message` property**
+
+  Add a new `Message` property to the `runtime.GlobalTestConfig`.
+  
+  ```go
+  type GlobalTestConfig struct {
+  	Env        map[string]string
+  	[...]
+  	Message    string
+  }
+  ```
+  
+  And last but not least implement the assignment of the global config. 
+  This can be done easily inside the `return` of the `ParseYAML` function statement.
+
+   ```go       
+  	return Suite{
+  		TestCases: tests,
+  		Config: runtime.TestConfig{
+  			InheritEnv: yamlConfig.Config.InheritEnv,
+  			Env:        yamlConfig.Config.Env,
+  			Dir:        yamlConfig.Config.Dir,
+  			Timeout:    yamlConfig.Config.Timeout,
+  			Retries:    yamlConfig.Config.Retries,
+  			Interval:   yamlConfig.Config.Interval,
+  			Nodes:      yamlConfig.Config.Nodes,
+  		},
+  		Nodes: convertNodes(yamlConfig.Nodes),
+  	}
+   ```
+  
+**6. Run the tests**
+
+  ```bash
+  $ make test
+  INFO: Starting build test
+  go test ./...
+  ok      github.com/SimonBaeumer/commander/cmd/commander 0.011s
+  ok      github.com/SimonBaeumer/commander/pkg/app       0.014s
+  ok      github.com/SimonBaeumer/commander/pkg/matcher   (cached)
+  ok      github.com/SimonBaeumer/commander/pkg/output    (cached)
+  ok      github.com/SimonBaeumer/commander/pkg/runtime   0.229s
+  ok      github.com/SimonBaeumer/commander/pkg/suite     0.008s
+  ```
+     
+**7. Learning by doing**
```

Comment:
I thought this was very well done 👍. Next time maybe a mention of suggestion of `learn by doing` at the top? Other than that this was very in depth. This will definitely save many people time.

Reply:
Upadted the title of the section to `Add a new field to the `YAML` suite - with a little leaning-by-doing task`